compiler/src/main/scala/org/scalajs/nscplugin/PrepJSInterop.scala

```diff
@@ -1312,36 +1296,47 @@ abstract class PrepJSInterop[G <: Global with Singleton](val global: G)
       }
     }
 
-    private def checkJSNameAnnots(sym: Symbol): Unit = {
-      for (annot <- sym.getAnnotation(JSNameAnnotation)) {
-        // Check everything about the first @JSName annotation
-        if (sym.isLocalToBlock || (enclosingOwner isnt OwnerKind.JSType)) {
-          reporter.error(annot.pos,
-              "@JSName can only be used on members of JS types.")
-        } else if (sym.isTrait) {
-          reporter.error(annot.pos,
-              "@JSName cannot be used on traits.")
-        } else if ((sym.isMethod || sym.isClass) && isPrivateMaybeWithin(sym)) {
-          reporter.error(annot.pos,
-              "@JSName cannot be used on private members.")
-        } else {
-          if (shouldCheckLiterals)
-            checkJSNameArgument(sym, annot)
-        }
-      }
-    }
+    private def checkJSCallingConventionAnnots(sym: Symbol): Unit = {
+      val callingConvAnnots = sym.annotations.filter(annot => JSCallingConventionAnnots.contains(annot.symbol))
 
-    private def checkDuplicateJSMemberAnnots(sym: Symbol): Unit = {
-      sym.annotations
-        .filter(annot => JSMemberAnnots.contains(annot.symbol))
-        .drop(1)
-        .foreach { annot =>
-          reporter.error(annot.pos, "A member can have at most one " +
-              "annotation among @JSName, @JSBracketAccess and @JSBracketCall.")
-        }
+      callingConvAnnots match {
+        case Nil =>
+          () // OK
+
+        case annot :: rest =>
+          def annotName: String = annot.symbol.nameString
+
+          if (sym.isLocalToBlock || (enclosingOwner isnt OwnerKind.JSType)) {
+            reporter.error(annot.pos,
+                s"@$annotName can only be used on members of JS types.")
```

Comment:
I'm wondering if we should list the annotations here statically (so that we explain the entire rule to the user, not just the part they are violating).

OTOH, for this specific case, it might indeed be unnecessarily wordy.

Reply:
I think it's better to keep it short in this case. It's unlikely that explaining the whole rule will help the user to fix their issue in this case.